server/pulp/server/webservices/views/users.py

```diff
@@ -0,0 +1,182 @@
+                  'name': name}
+
+        user = manager.create_user(*args, **kwargs)
+
+        # Add the link to the user
+        link = add_link(user)
+
+        # Grant permissions
+        permission_manager = factory.permission_manager()
+        permission_manager.grant_automatic_permissions_for_resource(link['_href'])
+
+        response = generate_json_response_with_pulp_encoder(user)
+        return generate_redirect_response(response, link['_href'])
+
+
+class UserResourceView(View):
+    """
+    View for a specific user.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, resuest, login):
+        """
+        Retrieve a specific user.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param login: login for the requested user
+        :type login: str
+
+        :return: Response containing the user
+        :rtype: django.http.HttpResponse
+        """
+        user = factory.user_query_manager().find_by_login(login)
+        if user is None:
+            raise pulp_exceptions.MissingResource(login)
+
+        add_link(user)
+        process_dictionary_against_whitelist(user, USER_WHITELIST)
+        return generate_json_response_with_pulp_encoder(user)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, login):
+        """
+        Delete a user.
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param login: login for the requested user
+        :type login: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+        manager = factory.user_manager()
+        result = manager.delete_user(login)
+
+        # Delete any existing user permissions given to the creator of the user
+        link = {'_href': reverse('user_resource',
+                kwargs={'login': login})}
+        if Permission.get_collection().find_one({'resource': link['_href']}):
+            Permission.get_collection().remove({'resource': link}, safe=True)
+
+        return generate_json_response(result)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def put(self, request, login):
+        """
+        Update a user.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param login: login for the requested user
+        :type login: str
+
+        :return: Response containing the user
+        :rtype: django.http.HttpResponse
+        """
+        # Pull all the user update data
+        user_data = request.body_as_json
+        delta = user_data.get('delta', None)
+
+        # Perform update
+        manager = factory.user_manager()
+        result = manager.update_user(login, delta)
+        process_dictionary_against_whitelist(result, USER_WHITELIST)
+        return generate_json_response_with_pulp_encoder(result)
+
+
+def add_link(user):
```

Comment:
This block is getting coverage from the fact that it is being used black box style in the other tests, but I think this should also get its own unit test.